message/html/markdown.js

```diff
@@ -15,29 +15,33 @@ exports.create = function (api) {
 
   function markdown (content) {
     if (typeof content === 'string') { content = {text: content} }
-    // handle patchwork style mentions.
+    // handle patchwork style mentions and custom emoji.
     var mentions = {}
     if (Array.isArray(content.mentions)) {
       content.mentions.forEach(function (link) {
-        if (link.name) mentions['@' + link.name] = link.link
+        if (link && link.name) mentions[link.name] = link.link
       })
     }
 
     var md = h('div', {className: 'Markdown'})
     md.innerHTML = renderer.block(content.text, {
-      emoji: renderEmoji,
+      emoji: (emoji) => {
+        var url = mentions[emoji]
+          ? api.blob.sync.url(mentions[emoji])
+          : api.emoji.sync.url(emoji)
+        return renderEmoji(emoji, url)
+      },
       toUrl: (id) => {
         if (ref.isBlob(id)) return api.blob.sync.url(id)
-        return (mentions[id] ? mentions[id] : id)
+        return mentions[id.substr(1)] || id
```

Comment:
why `substr(1)` in this case but not before? i think i understand, but would be good to double check and maybe leave a comment.